### PR TITLE
Add more text to the readme

### PR DESCRIPTION
**Pull request by rklaehn**
I think this should be good enough to give people familiar with ipfs a good idea how this works.

This is not meant to be an introduction to ipfs, so I won't explain pinning etc.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+blocks.
+
+GC will run incrementally, deleting blocks until the size targets are no longer exceeeded. The
+order in which unpinned blocks will be deleted can be customized.
+
+## Caching
+
+For unpinned blocks, it is possible to customize which blocks have the highest value. 
+
+<!-- cargo-sync-readme end -->
```
Comment (wngr) [inline README.md:43]: .. or you just omit that stuff in the readme and refer to the docs. What would be nice here though is some usage snippet.

File: src/lib.rs [hunk 2]
```diff
@@ -1,3 +1,35 @@
...
+//! to by the root will be collected by gc. However, a root being aliased does not mean that the dag
+//! must be complete.
+//!
+//! ## Temporary aliases
+//!
+//! A temporary alias is an unnamed alias that is just for the purpose of protecting blocks from gc
+//! while a large tree is being constructed. While an alias maps a single name to a single root, a
+//! temporary alias can be assigned to an arbitary number of blocks before the dag is finished.
+//!
+//! A temporary alias will be deleted as soon as the handle goes out of scope.
```
Comment (wngr) [inline src/lib.rs:19]: link to the struct + usage in doc comment there would be nice

File: src/lib.rs [hunk 3]
```diff
@@ -1,3 +1,35 @@
...
+//!
+//! ## GC
+//!
+//! GC refers to the process of removing unpinned blocks. It runs only when the configured size
+//! targets are exceeded. Size targets contain both the total size of the store and the number of
+//! blocks.
+//!
+//! GC will run incrementally, deleting blocks until the size targets are no longer exceeeded. The
```
Comment (wngr) [inline src/lib.rs:27]:
```suggestion
//! GC will run incrementally, deleting blocks until the size targets are no longer exceeded. The
```

File: src/lib.rs [hunk 4]
```diff
@@ -1,3 +1,35 @@
...
+//! order in which unpinned blocks will be deleted can be customized.
+//!
+//! ## Caching
+//!
+//! For unpinned blocks, it is possible to customize which blocks have the highest value.
```
Comment (wngr) [inline src/lib.rs:32]: highest value? So there's some kind of heuristic to determine which unpinned blocks to keep?

File: src/lib.rs [hunk 5]
```diff
@@ -1,3 +1,35 @@
...
+//!
+//! ## Temporary aliases
+//!
+//! A temporary alias is an unnamed alias that is just for the purpose of protecting blocks from gc
+//! while a large tree is being constructed. While an alias maps a single name to a single root, a
+//! temporary alias can be assigned to an arbitary number of blocks before the dag is finished.
+//!
+//! A temporary alias will be deleted as soon as the handle goes out of scope.
+//!
+//! ## GC
```
Comment (wngr) [inline src/lib.rs:21]:
```suggestion
//! ## GC (Garbage Collection)
```

File: src/lib.rs [hunk 6]
```diff
@@ -1,3 +1,35 @@
...
+//!
+//! GC refers to the process of removing unpinned blocks. It runs only when the configured size
+//! targets are exceeded. Size targets contain both the total size of the store and the number of
+//! blocks.
```
Comment (wngr) [inline src/lib.rs:25]: Doesn't it run on a fixed interval?!

Comment (rklaehn) [reply src/lib.rs:25]: It checks if the size targets are exceeded at a fixed interval, but if they are not it will not do anything.

Comment (wngr) [reply src/lib.rs:25]: I see. Then maybe add that here please.

File: README.md (hunk 1 above)
Comment (rklaehn) [reply README.md]: I don't think I want a usage snippet. It will get out of date soon since this is not done yet. But I dropped the sync-readme thing.

**Review by wngr — APPROVED** [review]

**Pull request merged**